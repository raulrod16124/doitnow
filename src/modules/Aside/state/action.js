import { asideTypes } from "./type";

export const HomeViewsVisibility = (view) => {
  const typeofView = view
    ? asideTypes.homeTodayView
    : asideTypes.homeTimestampView;
  return {
    type: typeofView,
    payload: view,
  };
};
