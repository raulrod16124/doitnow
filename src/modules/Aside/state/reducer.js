import { asideTypes } from "./type";

export const AsideReducer = (state = false, action) => {
  switch (action.type) {
    case asideTypes.homeTodayView:
      state = action.payload;
      return state;
    case asideTypes.homeTimestampView:
      state = action.payload;
      return state;
    default:
      return state;
  }
};
