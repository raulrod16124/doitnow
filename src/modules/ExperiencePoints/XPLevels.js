export const XPLevels = (totalXP) => {
  let diff = 500;
  let level = 1;
  while (totalXP >= diff) {
    level++;
    totalXP -= diff;
    diff += 100;
  }
  return { level, diff };
};

export const handleCalcutlateXP = (list) => {
  let easyTasks = 0;
  let mediumTasks = 0;
  let hardTasks = 0;
  let noAssigned = 0;

  list.map((task) => {
    if (task.status === "done" || task.status === "archive") {
      switch (task.level) {
        case "easy":
          easyTasks++;
          break;
        case "medium":
          mediumTasks++;
          break;
        case "hard":
          hardTasks++;
          break;
        default:
          noAssigned++;
          break;
      }
    }
  });
  const totalXP = easyTasks * 50 + mediumTasks * 75 + hardTasks * 100;
  return totalXP;
};

export const NextLevelXP = (level) => {
  let diff = 500;
  let xpPoints = 0;
  let count = 0;
  while (level > count) {
    count++;
    xpPoints += diff;
    diff += 100;
  }
  return xpPoints;
};
