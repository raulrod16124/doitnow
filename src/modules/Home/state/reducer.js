import TODOS from './../provider/todos.json';
import { TYPES } from "./type";

const initialState = [
    {
        id: "1111",
        title: "BOOM",
        level: "easy",
        status:"todo",
        description: "nothing",
    }
]

export const TodosReducer = ( state = initialState, action ) => {
    
    switch (action.type) {
        case TYPES.getTodo:
            return state
        case TYPES.createTodo:
            return [...state, action.payload];
        case TYPES.updateTodo:
            console.log(action.payload.status);
            state[action.payload.id] = action.payload;
            return {...state};
        case TYPES.deleteTodo:
            delete state[action.payload];
            return {...state}; 
    
        default:
            return state;
    }
}