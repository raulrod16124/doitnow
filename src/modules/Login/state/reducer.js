import { TYPES } from "./type";

const initialState = {
  status: "initial",
  error: "",
  data: {},
};

export const LoginReducer = (state = initialState, action) => {
  switch (action.type) {
    case TYPES.checkUser:
      let userData = null;
      // // console.log(action.payload);
      if (action.payload.user?.stsTokenManager.refreshToken) {
        userData = {
          id: action.payload._tokenResponse.localId,
          token: action.payload._tokenResponse.idToken,
          email: action.payload.user.providerData[0].email,
          name: action.payload.user.providerData[0].email.split("@")[0],
        };
      }
      if (userData?.token) {
        return {
          ...state,
          data: userData,
          status: "success",
        };
      } else {
        return {
          ...state,
          error: action.payload.split("/")[1].replace("-", " "),
          status: "error",
        };
      }

    case TYPES.createNewUser:
      // console.log(action.payload);
      if (action.payload.email) {
        return {
          ...state,
          data: action.payload,
          status: "created_account",
        };
      } else {
        return {
          ...state,
          error: action.payload.split("/")[1].replace("-", " "),
          status: "error",
        };
      }
    // TODO - Refactor to use firebase
    case TYPES.updateUser:
    // // console.log(action.payload);
    // const updateTodo = state.map( todo => {
    //     if( todo.id === action.payload.id ){
    //         return todo = action.payload;
    //     };
    //     return todo;
    // } );
    // return updateTodo;
    case TYPES.deleteUser:
    // // console.log(action.payload);
    // const updateTodos = state.filter( todo => todo.id !== action.payload );
    // return updateTodos;

    default:
      return state;
  }
};
