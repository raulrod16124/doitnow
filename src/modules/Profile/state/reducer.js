import { TYPES } from "./type";

const initialState = {
  status: "initial",
  error: "",
  data: [],
};

export const ProfileReducer = (state = initialState, action) => {
  switch (action.type) {
    case TYPES.getUserProfile:
      console.log(action.payload);
      if (action.payload) {
        return {
          ...state,
          data: action.payload,
          status: "success",
        };
      } else {
        return {
          ...state,
          error: action.payload,
          status: "error",
        };
      }
    case TYPES.createUserProfile:
      console.log(action.payload);
      if (action.payload) {
        return {
          ...state,
          data: action.payload,
          status: "created profile success",
        };
      } else {
        return {
          ...state,
          error: action.payload,
          status: "error",
        };
      }
    case TYPES.updateUserProfile:
      if (action.payload) {
        return {
          ...state,
          status: "userData_updated",
        };
      } else {
        return {
          ...state,
          error: action.payload,
          status: "error",
        };
      }

    default:
      return state;
  }
};
