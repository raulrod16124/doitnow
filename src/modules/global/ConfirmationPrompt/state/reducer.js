const initialState = {
  status: "close",
  acceptButton: "Accept",
  cancelButton: "Cancel",
  handleReject: null,
  handleAccept: null,
  message: null,
};

export const ConfirmationPromptReducer = (state = initialState, action) => {
  switch (action.type) {
    case "@open_confirmation_propmt":
      // console.log("OPEN POPUP");
      return { ...state, status: "open", ...action.payload };
    case "@close_confirmation_propmt":
      // console.log("CLOSE POPUP");
      return initialState;
    default:
      return state;
  }
};
