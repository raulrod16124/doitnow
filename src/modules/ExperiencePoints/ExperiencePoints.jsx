import React, { useEffect, useState } from "react";
import { useDispatch } from "react-redux";
import { useSelector } from "react-redux";

import { Button } from "../../stories/Button";
import { GetTasks } from "../Home/state/actions";
import { GetUserProfile, UpdateUserProfile } from "../Profile/state/actions";
import { defaultAvatar } from "../Profile/views/components/avatars";
import { XPLevels, handleCalcutlateXP } from "./XPLevels";

export const ExperiencePoints = () => {
  const todosState = useSelector((state) => {
    return state.TodosReducer;
  });

  const profileState = useSelector((state) => {
    return state.ProfileReducer;
  });

  const dispatch = useDispatch();

  const [profileData, setProfileData] = useState(null);

  const [xpVisibility, setXpVisibility] = useState(false);

  const [expereincePoints, setExperiencePoints] = useState(0);

  const [prevLevel, setprevLevel] = useState(0);

  const [actualLevel, setActualLevel] = useState(0);

  useEffect(async () => {
    const userData = await JSON.parse(
      localStorage.getItem(process.env.REACT_APP_LOCAL_STORAGE_KEY)
    );
    if (
      todosState.status === "task_created" ||
      todosState.status === "task_deleted" ||
      todosState.status === "task_updated"
    ) {
      setXpVisibility(false);
      dispatch(GetTasks(userData));
      dispatch(GetUserProfile(userData.id));
    }
    if (todosState.status === "success") {
      const totalXpPoints = handleCalcutlateXP(todosState.data);
      setExperiencePoints(totalXpPoints);
    }
  }, [todosState]);

  useEffect(() => {
    if (profileState.status === "success") {
      setProfileData(profileState.data);
    }
  }, [profileState]);

  useEffect(async () => {
    const { level } = XPLevels(expereincePoints);
    if (profileData && profileData.level < level) {
      // console.log("LEVEL UP GOOO");
      setprevLevel(profileData.level);
      setActualLevel(level);
      setXpVisibility(true);
      dispatch(
        UpdateUserProfile(profileData.id, {
          ...profileData,
          level: level,
        })
      );
    }
    if (profileData && profileData.level > level) {
      dispatch(
        UpdateUserProfile(profileData.id, {
          ...profileData,
          level: level,
        })
      );
    }
    // console.log("EXP POINTS CHECKED");
  }, [expereincePoints]);

  return (
    <>
      {xpVisibility && (
        <div
          className="bg-experience-points-window"
          onClick={(e) => {
            if (e.target.className === "bg-experience-points-window") {
              setXpVisibility(false);
            }
          }}
        >
          <div className="experience-points-window">
            <div className="content-avatar">
              <img
                className="avatar"
                src={
                  profileData && profileData.avatar
                    ? profileData.avatar
                    : defaultAvatar
                }
                alt="avatar"
              />
            </div>
            <h2 className="level-title">Level Up</h2>
            <div className="transition-level">
              <p className="prev-level">{prevLevel}</p>
              <i className="fas fa-chevron-right icon"></i>
              <p className="actual-level">{actualLevel}</p>
            </div>
            <div className="button-content">
              <Button
                label="Keep working hard"
                primary
                onClick={() => setXpVisibility(false)}
              />
            </div>
          </div>
        </div>
      )}
    </>
  );
};
