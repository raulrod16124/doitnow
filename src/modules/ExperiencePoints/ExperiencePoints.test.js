import { render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import { applyMiddleware, combineReducers, createStore } from "redux";
import thunk from "redux-thunk";

import { TodosReducer } from "../Home/state/reducer";
import { ProfileReducer } from "../Profile/state/reducer";
import { ExperiencePoints } from "./ExperiencePoints";

describe("Testing ExperiencePoints component", () => {
  let mockStore;
  let wrapper;

  beforeEach(() => {
    const reducers = combineReducers({
      TodosReducer: TodosReducer,
      ProfileReducer: ProfileReducer,
    });
    mockStore = createStore(reducers, applyMiddleware(thunk));
    wrapper = render(
      <Provider store={mockStore}>
        <MemoryRouter>
          <ExperiencePoints booleanForTest={true} />
        </MemoryRouter>
      </Provider>
    );
  });

  test("should render an ExperiencePoints component correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should not find role avatar until", () => {
    screen.debug();
  });
});
