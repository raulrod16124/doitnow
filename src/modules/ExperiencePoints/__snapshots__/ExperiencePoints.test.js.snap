// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ExperiencePoints component should render an ExperiencePoints component correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="bg-experience-points-window"
      >
        <div
          class="experience-points-window"
        >
          <div
            class="content-avatar"
          >
            <img
              alt="avatar"
              class="avatar"
              src="https://cdn-icons-png.flaticon.com/128/616/616412.png"
            />
          </div>
          <h2
            class="level-title"
          >
            Level Up
          </h2>
          <div
            class="transition-level"
          >
            <p
              class="prev-level"
            >
              0
            </p>
            <i
              class="fas fa-chevron-right icon"
            />
            <p
              class="actual-level"
            >
              0
            </p>
          </div>
          <div
            class="button-content"
          >
            <button
              class="sc-bdnxRM gKMDmj"
              type="submit"
            >
              Keep working hard
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="bg-experience-points-window"
    >
      <div
        class="experience-points-window"
      >
        <div
          class="content-avatar"
        >
          <img
            alt="avatar"
            class="avatar"
            src="https://cdn-icons-png.flaticon.com/128/616/616412.png"
          />
        </div>
        <h2
          class="level-title"
        >
          Level Up
        </h2>
        <div
          class="transition-level"
        >
          <p
            class="prev-level"
          >
            0
          </p>
          <i
            class="fas fa-chevron-right icon"
          />
          <p
            class="actual-level"
          >
            0
          </p>
        </div>
        <div
          class="button-content"
        >
          <button
            class="sc-bdnxRM gKMDmj"
            type="submit"
          >
            Keep working hard
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
